### PR TITLE
exit with non-zero status if warnings are present

### LINKED ISSUE #20: Non-zero exit code on found problems
I checked statix today and except for the unquoted splices (#8) and empty pattern I agree with its suggestions on my projects and I'd like to add it to CI.

Though to make any use of it in CI, it is required that we get a non-zero exit code on found suggestions.

### PR DESCRIPTION
**Pull request by nerdypepper**
should fix #20

### CODE REVIEW

**Comment by m1-s** [comment]
tested and worked as expected.

By the way the readme says to build with `nix build git+https://git.peppe.rs/languages/statix` but this did not work for me. Maybe it should say `nix build git+https://github.com/nerdypepper/statix.git`. Also to build this specific branch i used `nix build github:nerdypepper/statix/feat/exit-code`

File: bin/src/lint.rs [hunk 1]
```diff
@@ -54,17 +54,20 @@ pub mod main {
...
+        let results = vfs.iter().map(lint).collect::<Vec<_>>();
+
+        if results.len() != 0 {
+            for r in &results {
+                stdout.write(&r, &vfs, check_config.format).unwrap();
+            }
+            std::process::exit(1);
+        }
+
+        std::process::exit(0);
```
Comment (tfc) [inline bin/src/lint.rs:71]: minor nitpick, why not:

```suggestion
        for r in &results {
            stdout.write(&r, &vfs, check_config.format).unwrap();
        }

        std::process::exit(if results.is_empty() {0} else {1});
```

Comment (nerdypepper) [reply bin/src/lint.rs:71]: I do like that, thanks!

**Comment by nerdypepper** [comment]
@m1-s strange, this: `nix build git+https://git.peppe.rs/languages/statix` works fine for me. What is the exact error? Could it have been some kind of spurious network error? Could you open another issue with the same?

**Pull request merged**